src/Controller.cpp

```diff
@@ -84,7 +84,30 @@ void Controller::setFrameHandler(void (*fptr)(CANFrame *msg), byte opcodes[], by
 //
 void Controller::setParams(unsigned char *mparams)
 {
+  bool producer = 0;
+  bool consumer = 0;
+  bool nPreNormal = 0;
+  bool conOwn = 0;
+  byte mode = module_config->currentMode;
+  
   _mparams = mparams;
+  
+  for (Service * svc : services)
+  {
+    if (svc->getServiceID() == 5)
+    {
+      producer = 1;
+    }
+    if (svc->getServiceID() == 6)
+    {
+      consumer = 1;
+    }
+  }
+  if ((mode != MODE_UNINITIALISED) && (mode != MODE_SETUP))
```

Comment:
The persisted currentMode cannot be MODE_SETUP, right?

Reply:
Correct!  This can be much simplified